src/main/java/frc/robot/Constants.java

```diff
@@ -32,8 +37,27 @@ public static final class Drive {
         public static final double slowRotationMultiplier = Units.feetToMeters(15);
         public static final double fastTranslationMultiplier = Units.feetToMeters(16.6);
         public static final double fastRotationMultiplier = Units.feetToMeters(25);
-    }
+        
+        public static final Pose2d ampSpeakerPose = 
```

Comment:
DriverStation alliance is not always set correctly on robot startup. For instance, if you start up the robot before you plug into the field, this value won't be set and the robot code will crash. You need to save both poses as constants, and then do the ternary in the code at runtime.